pkg/k8shandler/util.go

```diff
@@ -9,10 +9,34 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	api "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	OsNodeLabel = "kubernetes.io/os"
+	LinuxValue  = "linux"
+)
+
+// EnsureLinuxNodeSelector takes given selector map and returns a selector map with linux node selector added into it.
+// If there is already a node type selector and is different from "linux" then it is overridden and warning is logged.
+// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+func EnsureLinuxNodeSelector(selectors map[string]string) map[string]string {
```

Comment:
golang uses capitalized function names to denote public vs private. I don't think we need to have this method be publicly available... can you make it `ensureLinuxNodeSelector` instead?

Reply:
Done. The original idea was that this function would be later called (re-used) from CLO as well. But we can make it public only once we get to that point in the future.